src/gui/screen_menu_esp_update.cpp

```diff
@@ -0,0 +1,332 @@
+};
+ESP_FLASHER::Msg ESP_FLASHER::msg = ESP_FLASHER::Msg::NoMsg;
+bool ESP_FLASHER::esp_connected = false;
+void ESP_FLASHER::Reset() {
+    esp_connected = false;
+    esp_loader_reset_target();
+    char tmp_str1[] = "- ESP not connected\n";
+    WriteStr(tmp_str1);
+}
+void ESP_FLASHER::Connect() {
+    char tmp_str[ESP_DESC_SIZE];
+    if (connect_to_target() == ESP_LOADER_SUCCESS) {
+        // msg = Msg::ConnectOK;
+        esp_connected = true;
+        char tmp_str1[] = "- ESP Connected\n";
+        strcpy(tmp_str, tmp_str1);
+        _dbg0("SYNC DONe - connected with ESP");
+    } else {
+        // msg = Msg::ConnectNOK;
+        char tmp_str1[] = "- ESP Connection FAILED\n";
+        strcpy(tmp_str, tmp_str1);
+        _dbg0("SYNC FAiLED");
+    }
+    WriteStr(tmp_str);
+}
+void ESP_FLASHER::FlashESP_AT() {
+    if (esp_connected) {
+        get_esp8266_binaries(&bin);
+        esp_loader_error_t err;
+        char tmp_str[ESP_DESC_SIZE];
+
+        // -- BLINK TEST FW (flashing on board LED)
+        strcpy(tmp_str, "- Flashing blink.bin\n");
+        WriteStr(tmp_str);
+        err = flash_binary(bin.boot.data, bin.boot.size, bin.boot.addr);
+        if (err == ESP_LOADER_SUCCESS) {
+            strcpy(tmp_str, "- Flash SUCCESS\n");
+        } else {
+            strcpy(tmp_str, "- Flash ERROR: Try again\n");
+        }
+        WriteStr(tmp_str);
+
+        // -- ESP AT FW v.1.7.4
+        // strcpy(tmp_str, "- Flashing boot1.7.bin\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.boot.data, bin.boot.size, bin.boot.addr);
+        //
+        // strcpy(tmp_str, "- Flashing user partition\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.user.data, bin.user.size, bin.user.addr);
+        //
+        // strcpy(tmp_str, "- Flashing blank.bin\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.blank1.data, bin.blank1.size, bin.blank1.addr);
+        //
+        // strcpy(tmp_str, "- Flashing init_data.bin\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.init_data.data, bin.init_data.size, bin.init_data.addr);
+        //
+        // strcpy(tmp_str, "- Flashing blank2.bin\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.blank2.data, bin.blank2.size, bin.blank2.addr);
+        //
+        // strcpy(tmp_str, "- Flashing blank3.bin\n");
+        // WriteStr(tmp_str);
+        // err = flash_binary(bin.blank3.data, bin.blank3.size, bin.blank3.addr);
+    }
+}
+uint8_t ESP_FLASHER::GetLinesInBuffer(char *str) {
+    // count lines
+    uint8_t i = 0;
+    uint8_t lines = 0;
+    char d;
+    d = str[0];
+    while (d != '\0') {
+        if (d == '\n') {
+            lines++;
+        }
+        i++;
+        d = str[i];
+    }
+    return lines;
+}
+void ESP_FLASHER::WriteStr(char *str) {
+    memset(plan_str, 0, ESP_DESC_SIZE);
+    strcpy(plan_str, str);
+
+    /*     char tmp_str1[ESP_DESC_SIZE]; */
```

Comment:
Does this commented code have some meaning (e.g. for future improvements)? If not it should be removed from the PR